pulser/utils.py

```diff
@@ -15,6 +15,36 @@
 import warnings
 
 
+def obj_to_dict(obj, *args, _build=True, _module=None, _name=None, **kwargs):
+    """Encodes an object in a dictionary for serialization.
+
+    Args:
+        obj: The object to encode in the dictionary.
+
+    Other parameters:
+        _build (bool): Whether the object is to be built on deserialization.
+        _module (str): Custom name for the module containing the object.
+        _name (str): Custom name of the object.
+        args: If the object is to be built, the arguments to give on creation.
+        kwargs: If the object is to be built, the keyword arguments to give on
+            creation.
+
+    Returns:
+        dict: The dictionary encoding the object.
+    """
+
+    d = {
+        "_build": _build,
```

Comment:
Just a thought, but since we are rebuilding all the object's internals just from its arguments, would it be worth having the version of pulser used ? Otherwise deserializing an object from one version to another version of pulser might lead to unexpected behaviours.
Having the version on each obj might way waay to redundant though, I'm sure we can think of something smarter, at the root level of the object. The root level is always a sequence, right ?

Reply:
It's well thought. Indeed, the root level is always a Sequence, so we probably can "teach" the decoder to extract the version from there. From this point, if the version there does not match the running program's, can we change it and construct the object accordingly, or do we just raise an error?

Reply:
I'm not sure we can change the version like that.
I think we might have to keep a list of versions compatible with the current one, and if the client uses another version we might have to throw an error. Maintaining a list sounds a bit cumbersome... There must be some better way

Reply:
Yeah, works for me :+1: